File: NewDoctorMan/WebContent/Components/doctor.js
$(document).ready(function() {
	if ($("#alertSuccess").text().trim() == "") {
		$("#alertSuccess").hide();
	}
	$("#alertError").hide();
});

// SAVE ============================================
$(document).on("click", "#btnSave", function(event) {

	// Clear status msges-------------
	$("#alertSuccess").text("");
	$("#alertSuccess").hide();
	$("#alertError").text("");
	$("#alertError").hide();

	// Form validation----------------
	var status = validateItemForm();

	// If not valid-------------------
	if (status != true) {
		$("#alertError").text(status);
		$("#alertError").show();
		return;
	}

	// If valid----------------------
	// $("#formDoctor").submit();
	// });

	var type = ($("#hidDoctorIDSave").val() == "") ? "POST" : "PUT";

	$.ajax({
		url : "DoctorAPI",
		type : type,
		data : $("#formDoctor").serialize(),
		dataType : "text",
		complete : function(response, status) {
			onDoctorSaveComplete(response.responseText, status);
		}

	});
});

function onDoctorSaveComplete(response, status) {
	if (status == "success") {
		var resultSet = JSON.parse(response);

		if (resultSet.status.trim() == "success") {
			$("#alertSuccess").text("Successfully saved.");
			$("#alertSuccess").show();

			$("#divDoctorsGrid").html(resultSet.data);
		} else if (resultSet.status.trim() == "error") {
			$("#alertError").text(resultSet.data);
			$("#alertError").show();
		}
	} else if (status == "error") {
		$("#alertError").text("Error while saving.");
		$("#alertError").show();
	} else {
		$("#alertError").text("Unknown error while saving..");
		$("#alertError").show();
	}

	$("#hidDoctorIDSave").val("");
	$("#formDoctor")[0].reset();
}

// UPDATE
$(document)
		.on("click", ".btnUpdate", function(event) {
					$("#hidDoctorIDSave").val(
							$(this).closest("tr").find('#hidDoctorIDUpdate')
									.val());
					$("#doctorName").val(
							$(this).closest("tr").find('td:eq(0)').text());
					$("#address").val(
							$(this).closest("tr").find('td:eq(1)').text());
					$("#phoneNum").val(
							$(this).closest("tr").find('td:eq(2)').text());
					$("#email").val(
							$(this).closest("tr").find('td:eq(3)').text());
					$("#gender").val(
							$(this).closest("tr").find('td:eq(4)').text());
					$("#age")
							.val($(this).closest("tr").find('td:eq(5)').text());
					$("#status").val(
							$(this).closest("tr").find('td:eq(6)').text());
					$("#specialization").val(
							$(this).closest("tr").find('td:eq(7)').text());
					$("#hospitalWork").val(
							$(this).closest("tr").find('td:eq(8)').text());
					$("#details").val(
							$(this).closest("tr").find('td:eq(9)').text());
					$("#username").val(
							$(this).closest("tr").find('td:eq(10)').text());
					$("#password").val(
							$(this).closest("tr").find('td:eq(11)').text());
				})

// REMOVE==========================================
$(document).on("click", ".btnRemove", function(event) {
	$.ajax({
		url : "DoctorAPI",
		type : "DELETE",
		data : "doctorID=" + $(this).data("doctorid"),
		dataType : "text",
		complete : function(response, status) {
			onDoctorDeleteComplete(response.responseText, status);
		}

	});
});

function onDoctorDeleteComplete(response, status) {
	if (status == "success") {
		var resultSet = JSON.parse(response);

		if (resultSet.status.trim() == "success") {
			$("#alertSuccess").text("Successfully deleted.");
			$("#alertSuccess").show();

			$("#divDoctorsGrid").html(resultSet.data);
		} else if (resultSet.status.trim() == "error") {
			$("#alertError").text(resultSet.data);
			$("#alertError").show();
		}
	} else if (status == "error") {
		$("#alertError").text("Error while deleting.");
		$("#alertError").show();
	} else {
		$("#alertError").text("Unknown error while deleting..");
		$("#alertError").show();
	}
}

// CLIENT-MODEL=================================================================

function validateItemForm() {

	// NAME
	if ($("#doctorName").val().trim() == "") {
		return "Insert doctor name.";
	}
	// ADDRESS
	if ($("#address").val().trim() == "") {
		return "Insert address.";
	}
	// PHONE NUMBER
	if ($("#phoneNum").val().trim() == "") {
		return "Insert phone number.";
	}
	// PHONE NUMBER NUMERICAL VALUE
	var phoneNum = $("#phoneNum").val().trim();
	if (!$.isNumeric(phoneNum)) {
		return "Insert a numerical value for phone number.";
	}
	
	//phone number validation 
	var contactNochar =/^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/;
	var contact = $("#phoneNum").val().trim();
	if (contactNochar.test(contact) == false) {
		return "Should enter valid Contact No";
	}
	console.log(contactNochar.test(contact));

	// E-MAIL
	if ($("#email").val().trim() == "") {
		return "Insert E-mail.";
	}
	
	//valid email characters
	var emailchar = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
	var email = $("#email").val().trim();
	if (emailchar.test(email) == false) {
		return "Should enter valid Email Address";
	}
	console.log(emailchar.test(email));

	// GENDER

	if ($("#gender").val() == "0") {
		return "Select a gender";
	}

	// AGE
	if ($("#age").val().trim() == "") {
		return "Insert age.";
	}

	// AGE NUUMERICAL VALUE
	var age = $("#age").val().trim();
	if (!$.isNumeric(age)) {
		return "Insert a numerical value for age.";
	}

	// STATUS

	if ($("#status").val() == "0") {
		return "Select a status";
	}

	// SPECIALIZATION
	if ($("#specialization").val().trim() == "") {
		return "Insert specialization.";
	}
	// HOSPITAL WORK
	if ($("#hospitalWork").val().trim() == "") {
		return "Insert hospital work.";
	}
	// Details
	if ($("#details").val().trim() == "") {
		return "Insert details.";
	}
	// USERNAME
	if ($("#username").val().trim() == "") {
		return "Insert username.";
	}
	
	//User validation
//	var usernamechar =/^[a-zA-Z0-9]+$/;
//	var username = $("#username").val().trim();
//	if (usernamechar.test(username) == false) {
//		return "Should enter valid Username";
//	}
//	console.log(usernamechar.test(username));
	
	
	// PASSWORD
	if ($("#password").val().trim() == "") {
		return "Insert password.";
	}
	
	//Password validation
//	var passwordchar =/^[a-zA-Z0-9!@#$%^&*]{6,16}$/;
//	var password = $("#password").val().trim();
//	if (passwordchar.test(password) == false) {
//		return "Should enter valid password";
//	}
//	console.log(passwordchar.test(password));
	
	return true;
}

File: NewDoctorMan/src/model/Doctor.java
package model;

import java.sql.*;

public class Doctor {
	// A common method to connect to the DB
	private Connection connect() {
		Connection con = null;
		try {
			Class.forName("com.mysql.jdbc.Driver");

			// Provide the correct details: DBServer/DBName, username, password
			con = DriverManager.getConnection("jdbc:mysql://127.0.0.1:3306/paf", "root", "");
		} catch (Exception e) {
			e.printStackTrace();
		}
		return con;
	}

	public String insertItem(String doctorName, String address, String phoneNum, String email, String gender, String age, String status, String specialization, String hospitalWork, String details, String username, String password)  
	{
		String output = ""; 
	 
	  try  
	  {
		  Connection con = connect(); 
	 
	   if (con == null)  
	   {
		   return "Error while connecting to the database for inserting."; 
	   } 
	 
	   // create a prepared statement    
	   String query = " insert into doctor(`doctorID`,`doctorName`,`address`,`phoneNum`,`email`,`gender`,`age`,`status`,`specialization`,`hospitalWork`,`details`,`username`,`password`)" + " values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"; 
	 
	   PreparedStatement preparedStmt = con.prepareStatement(query); 
	 
	   // binding values   
	   preparedStmt.setInt(1, 0);    
	   preparedStmt.setString(2, doctorName);    
	   preparedStmt.setString(3, address);    
	   preparedStmt.setString(4, phoneNum);
	   preparedStmt.setString(5, email);
	   preparedStmt.setString(6, gender);
	   preparedStmt.setString(7, age);
	   preparedStmt.setString(8, status);
	   preparedStmt.setString(9, specialization);
	   preparedStmt.setString(10, hospitalWork);
	   preparedStmt.setString(11, details);
	   preparedStmt.setString(12, username);
	   preparedStmt.setString(13, password);
	   
	// execute the statement    
	   preparedStmt.execute();    
	   con.close(); 
	   
	   output = "Inserted successfully";   
	   }   catch (Exception e)  
	  	
	  {   
		   output = "Error while inserting the item.";    
		   System.err.println(e.getMessage());  
		   
	  	} 
	 
	  return output;  
	  
	}

	public String readDoctors() {
		String output = "";
		try {
			Connection con = connect();
			if (con == null) {
				return "Error while connecting to the database for reading.";
			}
			// Prepare the html table to be displayed
			output = "<table border=\"1\"><tr><th>Doc Name</th><th>Address</th><th>Phone</th><th>E-mail</th><th>Gender</th>"
					+ "<th>Age</th><th>Status</th><th>Specialization</th><th>Hospital</th><th>Details</th><th>User Name</th><th>Password</th><th>Update</th><th>Remove</th></tr>";
			String query = "select * from doctor";
			Statement stmt = con.createStatement();
			ResultSet rs = stmt.executeQuery(query);
			// iterate through the rows in the result set
			while (rs.next()) {
				String doctorID = Integer.toString(rs.getInt("doctorID"));
				String doctorName = rs.getString("doctorName");
				String address = rs.getString("address");
				String phoneNum = rs.getString("phoneNum");
				String email = rs.getString("email");
				String gender = rs.getString("gender");
				String age = Integer.toString(rs.getInt("age"));
				String status = rs.getString("status");
				String specialization = rs.getString("specialization");
				String hospitalWork = rs.getString("hospitalWork");
				String details = rs.getString("details");
				String username = rs.getString("username");
				String password = rs.getString("password");

				// Add into the html table
				// output += "<tr><td>" + doctorID + "</td>";
				output += "<tr><td><input id =\"hidDoctorIDUpdate\" name=\"hidDoctorIDUpdate\" type=\"hidden\" value=\"" + doctorID + "\">"
						+ doctorName + "</td>";
				//output += "<tr><td>" + doctorName + "</td>";
				output += "<td>" + address + "</td>";
				output += "<td>" + phoneNum + "</td>";
				output += "<td>" + email + "</td>";
				output += "<td>" + gender + "</td>";
				output += "<td>" + age + "</td>";
				output += "<td>" + status + "</td>";
				output += "<td>" + specialization + "</td>";
				output += "<td>" + hospitalWork + "</td>";
				output += "<td>" + details + "</td>";
				output += "<td>" + username + "</td>";
				output += "<td>" + password + "</td>";
				
				// buttons
				output += "<td><input name=\"btnUpdate\" type=\"button\" value=\"Update\" class=\"btnUpdate btn btn-secondary\"></td>"
						+ "<td><form method=\"post\" action=\"doctors.jsp\">"
						+ "<input name=\"btnRemove\" type=\"submit\" value=\"Remove\" class=\"btn btn-danger\">"
						+ "<input name=\"hidDoctorIDDelete\" type=\"hidden\" value=\"" 
						+ doctorID + "\">" + " </form></td></tr> ";
						//+ "<input name=\"doctorID\" type=\"hidden\" value=\"" + doctorID + "\">" + "</form></td></tr>";
			}
			con.close();
			// Complete the html table
			output += "</table>";
		} catch (Exception e) {
			output = "Error while reading the doctors.";
			System.err.println(e.getMessage());
		}
		return output;
	}

	public String updateItem(String doctorID, String doctorName, String address, String phoneNum, String email,
			String gender, String age, String status, String specialization, String hospitalWork, String details,
			String username, String password) {
		String output = "";
		try {
			Connection con = connect();
			if (con == null) {
				return "Error while connecting to the database for updating.";
			}
			// create a prepared statement
			String query = "UPDATE doctor SET doctorName=?,address=?,phoneNum=?,email=?,gender=?,age=?,status=?,specialization=?,hospitalWork=?,details=?,username=?,password=? WHERE doctorID=?";
			PreparedStatement preparedStmt = con.prepareStatement(query);
			// binding values
			preparedStmt.setString(1, doctorName);
			preparedStmt.setString(2, address);
			preparedStmt.setString(3, phoneNum);
			preparedStmt.setString(4, email);
			preparedStmt.setString(5, gender);
			preparedStmt.setString(6, age);
			preparedStmt.setString(7, status);
			preparedStmt.setString(8, specialization);
			preparedStmt.setString(9, hospitalWork);
			preparedStmt.setString(10, details);
			preparedStmt.setString(11, username);
			preparedStmt.setString(12, password);
			preparedStmt.setInt(13, Integer.parseInt(doctorID));
			// execute the statement
			preparedStmt.execute();
			con.close();
			output = "Updated successfully";
		} catch (Exception e) {
			output = "Error while updating the doctor.";
			System.err.println(e.getMessage());
		}
		return output;
	}

	public String deleteItem(String doctorID) {
		String output = "";
		try {
			Connection con = connect();
			if (con == null) {
				return "Error while connecting to the database for deleting.";
			}
			// create a prepared statement
			String query = "delete from doctor where doctorID=?";
			PreparedStatement preparedStmt = con.prepareStatement(query);
			// binding values
			preparedStmt.setInt(1, Integer.parseInt(doctorID));
			// execute the statement
			preparedStmt.execute();
			con.close();
			output = "Deleted successfully";
		} catch (Exception e) {
			output = "Error while deleting the doctors.";
			System.err.println(e.getMessage());
		}
		return output;
	}

	
	
	//login
	public User readLogin(String username, String password) {
		try {
			
			Connection con = connect();
			
			if(con == null){
				
				System.out.println("executed up before");
				
				return null;
			}
			
			// create a prepared statement 
			
			String query = "select * from doctor where username = '" + username +"'";
			PreparedStatement preparedStmt = con.prepareStatement(query);
			Statement stmt = con.createStatement();
			ResultSet rs = stmt.executeQuery(query);
			
			while(rs.next()) {
				System.out.println(rs.getString("doctorID"));
				User u = new User(rs.getString("doctorID"), rs.getString("doctorName"), rs.getString("username"), "doctor");
				String pswd = rs.getString("password");
				
				if(pswd.contentEquals(password)) {
					
					System.out.println("true from db");
					
					return u;
					
				}else {
					System.out.println("false from db");
				}
			}
			
			con.close();
			
		}catch(Exception e) {
			System.out.println(e.getMessage());
		}
		
		return null;
	}
	
	
	//admin login
	
	public boolean readAdminLogin(String Admin_username, String Admin_password) {
		try {
			Connection con = connect();
				if(con == null){
				
				System.out.println("executed up before");
				
				return false;
			}
			
			// create a prepared statement 
			
			String query = "select Admin_password from admin_details where Admin_username = '" + Admin_username +"'";
			PreparedStatement preparedStmt = con.prepareStatement(query);
			Statement stmt = con.createStatement();
			ResultSet rs = stmt.executeQuery(query);
			
			while(rs.next()) {
				//System.out.println(rs.getString("Admin_id"));
	
				//User u = new User(rs.getString("Admin_id"), rs.getString("Admin_name"), rs.getString("Admin_username"), "Admin");
				String pswd = rs.getString("Admin_password");
			
				if(pswd.contentEquals(Admin_password)) {
					
					System.out.println("true from db");
					
					return true;
					
				}else {
					System.out.println("false from db");
				}
			}
			
			con.close();
			
		}catch(Exception e) {
			System.out.println(e.getMessage());
		}
		
		return false;
	}
	
}
